### ISSUE TITLE
BUG: gtk-text-attributes cstruct is incorrect

### ISSUE THREAD

**Issue by stacksmith** [crategus/cl-cffi-gtk#67]
See gtk/gtk.text-tag.lisp  - at the end, it is:
```
...
 (:invisible :uint)
  (:bg-full-height :uint)
  (:editable :uint))
```
while recent gtk documentation shows
```
  guint invisible : 1;
  guint bg_full_height : 1;
  guint editable : 1;
  guint no_fallback: 1;
  GdkRGBA *pg_bg_rgba;
  gint letter_spacing;
  gchar *font_features;
};
```
So there are 4 single-bit packed values, and 3 more are missing.

**Comment by Ferada** [comment]
Yeah I'm super annoyed with GTK using bitfields. Ideally those were supported in CFFI. If you really need it ... I suspect writing the wrapper in C is less work than supporting the correct sizes and alignment etc. in CFFI.

**Comment by stacksmith** [comment]
I believe layout of packed bitfields is not defined by the standard.  Compilers have flags that affect structure packing behavior, making the situation very unreliable.  I always assumed bitfields are packed into ints conceptually, but that is guesswork; in this case other members follow, which makes the offsets impossible to derive reliably.

**Comment by stacksmith** [comment]
I hate the idea of writing an extra C wrapper - the last time I went that route I wound up writing so much C that I was wondering why not just write the whole application in C...
I am working on a simple utility library that simplifies foreign structure access... It's a bit of a mess right now, but I will be cleaning it up as I go....

**Comment by Ferada** [comment]
Right, but all wrappers still won't allow us to access bitfields reliably, right?

**Comment by stacksmith** [comment]
I can't think of a compiler that does not align to nearest power-of-two for bytes and pack bitfields.  I suppose we should dig through the spec, although just thinking about it give mes the willies.  After a brief stint on X3J11, I never want to see C++ specs ever again.
